src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -54,6 +56,46 @@ public static Feature parseFeature(String featureName) throws ParseException {
         return new Feature(featureName);
     }
 
+    /**
+     * Parses {@code date} into a {@code AthletickDate} and returns it. Leading and trailing
+     * whitespaces will be trimmed.
+     * @throws ParseException if the specified date is invalid (not length of 6 or 8).
+     */
+    public static AthletickDate parseDate(String date) throws ParseException {
+        String trimmedDate = date.trim();
+        if (trimmedDate.length() == 6 || trimmedDate.length() == 8) {
+            SimpleDateFormat fullDate = new SimpleDateFormat("ddMMyyyy");
+            fullDate.setLenient(false);
+            SimpleDateFormat monthYear = new SimpleDateFormat("MMyyyy");
+            monthYear.setLenient(false);
+            try {
+                if (trimmedDate.length() == 8) {
+                    Date d = fullDate.parse(trimmedDate);
+                    int day = Integer.parseInt(new SimpleDateFormat("d").format(d));
+                    int month = Integer.parseInt(new SimpleDateFormat("M").format(d));
+                    int year = Integer.parseInt(new SimpleDateFormat("yyyy").format(d));
+                    int type = 1;
+                    String mth = new SimpleDateFormat("MMMM").format(d);
```

Comment:
What's this `MMMM` for? Is it part of month?

Reply:
MMMM formats date into the full string. my month variable stores the number e.g. 1 and mth stores the string e.g January